projects/cm_mcu/MonitorI2CTask.c

```diff
@@ -83,12 +83,15 @@ void MonitorI2CTask(void *parameters)
 
   bool good = false;
   for (;;) {
+
     log_debug(LOG_MONI2C, "%s: grab semaphore\r\n", args->name);
 
     // grab the semaphore to ensure unique access to I2C controller
-    if (acquireI2CSemaphore(args->xSem) == pdFAIL) {
-      log_warn(LOG_SERVICE, "%s could not get semaphore in time; continue\r\n", args->name);
-      continue;
+    if (args->xSem != NULL) {
+      if (acquireI2CSemaphore(args->xSem) == pdFAIL) {
+        log_warn(LOG_SERVICE, "%s could not get semaphore in time; continue\r\n", args->name);
+        continue;
+      }
     }
```

Comment:
Should we add a a `vTaskDelay` here? Otherwise, if the semaphore access fails, we just try again immediately in the next loop iteration.

Reply:
I have added vTaskDelay for 10ms before continue. That seems to be similar to the delay in MonitorTask.c [here](https://github.com/apollo-lhc/cm_mcu/blob/bugfix_3v8/projects/cm_mcu/MonitorTask.c#L109C9-L109C9), right?

If so, why is there so many parts in MON that require a delay by 10ms? should I modify MonitorTask the same as this part in MonitorI2CTask?  (i.e. just add a delay after semaphore failing at the time it attempts to get a semaphore for it to try again in the next loop iteration)

Reply:
let's create an issue. I prefer not to mash together too many changes. 
did you get a chance to test this on apollo205 as well as on the benchtop CM?